core/src/core/usb/types.zig

```diff
@@ -52,15 +52,27 @@ pub const ControlStage = enum {
 
 /// The types of USB SETUP requests that we understand.
 pub const SetupRequest = enum(u8) {
-    GetDescriptor = 0x06,
+    SetFeature = 0x03,
     SetAddress = 0x05,
+    GetDescriptor = 0x06,
     SetConfiguration = 0x09,
 
     pub fn from_u8(v: u8) ?@This() {
         return std.meta.intToEnum(@This(), v) catch null;
     }
 };
 
+pub const FeatureSelector = enum(u8) {
+    EndpointHalt = 0x00,
+    DeviceRemoteWakeup = 0x01,
+    TestMode = 0x02,
+    // The remaining features only apply to OTG devices.
+
+    pub fn from_u8(v: u8) ?@This() {
```

Comment:
Can't we just use `@enumFromInt`?

Reply:
Sort of. At first, I only did it like this to match the style of other decls in this file (`DescType`, `TransferType`, `SetupRequest`) as well as a few other flies. Those are all thin wrappers around `std.meta.intToEnum(...) catch null`.

But the main reason _not_ to use `@emunFromInt` directly is that the use case is decoding bytes from the host. With raw `@enumFromInt`, invalid bytes from the host would trigger illegal behavior rather than an error or null.

How would you like me to proceed? Happy to take it in any direction.